hazelcast-jet-core/src/main/java/com/hazelcast/jet/aggregate/AggregateOperations.java

```diff
@@ -705,15 +821,28 @@ private AggregateOperations() {
      * {@code HashMap} where the key is the result of applying {@code keyFn}
      * and the value is a list of the items with that key.
      * <p>
-     * This operation achieves the effect of a cascaded group-by where the
+     * Sample usage:
+     * <pre>{@code
+     *     // the result will be a single instance of HashMap<SensorId, List<Measurement>>
+     *     AggregateOperations.groupingBy(Measurement::getSensorId)
```

Comment:
The groupingBy aggregate operation seems quite confusing as one might use it instead of .groupingKey and lose the advantages of parallel aggregation. 
Could we make it explicit in the JavaDoc that .groupingKey should be always used before .groupingBy, including the reasoning?